airflow/jobs/scheduler_job.py

```diff
@@ -419,7 +419,7 @@ def manage_slas(self, dag: DAG, session: Session = None) -> None:
         ts = timezone.utcnow()
         for ti in max_tis:
             task = dag.get_task(ti.task_id)
-            if not isinstance(task.sla, timedelta):
+            if not task.sla:
```

Comment:
The problem with this approach is it will fail the entire DagfileProcessorProcess that is running SlaCallback.

So imagine if there are 5 tasks, only 1 of them has `task.sla=3` and others have `task.sla=timedelta(seconds=3)`, then with the current change it will raise type error and will not run or check sla for Tasks that have correct task.sla set.

Reply:
Yes, but in my opinion this is better than not knowing why SLA was not send. To avoid run time errors I think about adding a check to task constructor to fail as soon as possible - WDYT?

Reply:
I think a better fix would be something like:

```python
if task.sla and not isinstance(task.sla, timedelta):
     raise("Expected timedelta object, got %s instead: %s", type(task.sla), task.sla)
```

This would help users understand what's the problem